mmv1/third_party/terraform/services/compute/resource_compute_region_instance_group_manager.go.tmpl

```diff
@@ -1152,17 +1152,19 @@ func expandDistributionPolicy(d *schema.ResourceData) *compute.DistributionPolic
 		return nil
 	}
 
-	distributionPolicyZoneConfigs := make([]*compute.DistributionPolicyZoneConfiguration, 0, dpz.Len())
-	for _, raw := range dpz.List() {
-		data := raw.(string)
-		distributionPolicyZoneConfig := compute.DistributionPolicyZoneConfiguration{
-			Zone: "zones/" + data,
-		}
-
-		distributionPolicyZoneConfigs = append(distributionPolicyZoneConfigs, &distributionPolicyZoneConfig)
-	}
-
-	return &compute.DistributionPolicy{Zones: distributionPolicyZoneConfigs, TargetShape: dpts}
+  if d.HasChange("distribution_policy_zones") {
```

Comment:
It is better to not include the logic checking changes inside the function `func expandDistributionPolicy`. We prefer to separate those.

The field change checking for `distribution_policy_zones` should be put into `func resourceComputeRegionInstanceGroupManagerUpdate` around the lines [if d.HasChange("distribution_policy_target_shape")
](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/compute/resource_compute_region_instance_group_manager.go.tmpl#L909-L912)

Reply:
PTAL at my changes.
I think I understand your request - we use the same method in Create (where having a change seems weird), and in update.
My idea for the change is that since distribution_policy.zones are immutable, then we will never want to send them in the update request. The field is also marked as ForceNew, so if there is any change in the distribution_policy_zones, then we will destroy the resource and create it again.